File: api_nestjs_distributed_order_platform/src/application/port/product.repository.ts
import { ProductEntity } from "src/domain/entities/product.entity";

export interface ProductRepositoryInterface {
    create(product: ProductEntity): Promise<ProductEntity>;
    findById(productId: string): Promise<ProductEntity | null>;
    findAll(): Promise<ProductEntity[]>;
    update(product: ProductEntity): Promise<void>;
    delete(productId: string): Promise<void>;
    findBySku(sku: string): Promise<ProductEntity | null>;
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/port/ProductRepository.java
package com.project.order.application.port;

import com.project.order.domain.model.Product;

import java.util.List;
import java.util.Optional;

public interface ProductRepository {
    Product create(Product product);
    Optional<Product> findById(String id);
    Optional<Product> findBySku(String sku);
    List<Product> findAll();
    Product update(Product product);
    void delete(String id);
}
